src/wombats/game/processor.clj

```diff
@@ -103,23 +103,106 @@
     (.setPayload request payload)
     request))
 
+(defn- sh-handler
+  "This runs some code by creating a script and running it against an
+  interpreter. It feeds it the state on STDIN and reads the next move
+  from STDOUT.
+
+  If the subprocess throws an error, that's translated into an
+  exception."
+  [ext interpreter state code]
+  (let [tmp-file (java.io.File/createTempFile "bot" ext)
+        input (cheshire/generate-string state)]
+    (try
+      (with-open [file (clojure.java.io/writer tmp-file)]
+        (.write file code))
+      (let [proc (clojure.java.shell/sh interpreter (.getAbsolutePath tmp-file)
+                                        :in (java.io.StringReader. input))]
+        (if (zero? (:exit proc))
+          (cheshire/parse-string (:out proc) true)
+          (throw (Exception. (:err proc)))))
+      (finally
+        (.delete tmp-file)))))
+
+(defn- js-handler
+  "This evaluates the js bot code against the state given."
+  [code state timeout]
+  (sh-handler ".js" "node" state
+        (str "const stdin  = process.stdin,\n"
+             "      stdout = process.stdout;\n\n"
+             "const wombats = "
+             code
+             "\n\nconst timeLeft = " timeout "\n\n"
+             "var chunks = [];\n\n"
+             "stdin.resume();\n"
+             "stdin.setEncoding('utf8');\n\n"
+             "stdin.on('data', function(chunk) {\n"
+             "  chunks.push(chunk);\n"
+             "});\n\n"
+             "stdin.on('end', function() {\n"
+             "  var inputJSON  = chunks.join(),\n"
+             "      parsedData = JSON.parse(inputJSON),\n"
+             "      outputData = wombats(parsedData, timeLeft);\n\n"
+             "  stdout.write(JSON.stringify(outputData, null, '  '));\n"
+             "});\n")))
+
+(defn- clj-handler
+  "This evaluates the clj bot code against the state given."
+  [code state timeout]
+  ((load-string code) state timeout))
+
+(defn- py-handler
+  "This evalutes the py bot code against the state given."
+  [code state timeout]
+  (sh-handler ".py" "python" state
+        (str "import json, sys\n\n"
+             code
+             "\n"
+             "json.dump(wombat(json.load(sys.stdin), "
+             timeout
+             "), sys.stdout)\n")))
+
+(defn- dev-mode-request
+  [code path decision-maker-state]
+  (let [c (async/chan)
+        path-ext (keyword (last (clojure.string/split path #"\.")))
+        timeout 30000
+
+        handlers {:clj clj-handler
+                  :js js-handler
+                  :py py-handler}
+        handler (handlers path-ext)]
+
+    (future
+      (async/go (async/>! c (try
+                              {:response (handler code decision-maker-state timeout)
```

Comment:
I think that this gets handled correctly by the way I'm handling the interpreter scripts. Since the state gets written to STDIN on each script, the Python and JavaScript bots convert this as JSON. They write their output out, which gets interpreted using cheshire just as in the line you link to.

Or maybe I'm not getting something. I'll swing by your office, maybe Monday, to talk about it.